testing/integration/upgrade_test.go

```diff
@@ -121,17 +122,6 @@ func (s *UpgradeElasticAgent) TearDownTest() {
 	assert.NoError(s.T(), tools.UnEnrollAgent(s.requirementsInfo.KibanaClient))
 }
 
-func agentStatus(expectedStatus string, suite UpgradeElasticAgent) func() bool {
```

Comment:
This functionality was needed by the FQDN test as well, so I moved it to `tools.WaitForAgentStatus` and generalized it so any test suite could use it.